lib/HeroBonus.h

```diff
@@ -71,27 +71,27 @@ class CSelector : std::function<bool(const Bonus*)>
 class DLL_LINKAGE CBonusProxy
 {
 public:
-	CBonusProxy(const IBonusBearer * Target, CSelector Selector);
+	CBonusProxy(const IBonusBearer * Target, CSelector Selector, const std::string & cachingString = "");
 	CBonusProxy(const CBonusProxy & other);
 	CBonusProxy(CBonusProxy && other);
 
 	CBonusProxy & operator=(CBonusProxy && other);
 	CBonusProxy & operator=(const CBonusProxy & other);
-
-	TConstBonusListPtr get() const;
-
 	const BonusList * operator->() const;
-private:
-	mutable int64_t cachedLast;
-	const IBonusBearer * target;
+	TConstBonusListPtr getBonusList() const;
+
+protected:
 	CSelector selector;
-	mutable TConstBonusListPtr data;
+	std::string cachingStr; //for bonusList caching

```

Comment:
But without  `cachingStr`  caching does not work for objects of `CHeroWithMaybePickedArtifact` class because they are being created from scratch with all their proxies each time on user click by the game interface handlers. So, for hero window bonuses are taken without caching and for army stack with caching. It's strange, but it seems, it does not matter for UI, hardly anyone will notice the difference...

Reply:
Such objects never were a performance bottleneck.